subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -174,42 +240,48 @@ static int get_job_execution(struct mqtt_client *const client,
 
 	/* Check if message received is a job. */
 	err = aws_fota_parse_DescribeJobExecution_rsp(payload_buf, payload_len,
-						      job_id, hostname,
+						      job_id_incoming, hostname,
 						      file_path,
-						     &execution_version_number);
+						      &execution_version_number);
 
 	if (err < 0) {
 		LOG_ERR("Error when parsing the json: %d", err);
-		return err;
+		goto cleanup;
 	} else if (err == 0) {
 		LOG_DBG("Got only one field");
 		LOG_DBG("No queued jobs for this device");
 		return 0;
 	}
 
-	LOG_DBG("Job ID: %s", log_strdup(job_id));
+	/* Check if the incoming job is already being handled. */
+	if (strcmp(job_id_incoming, job_id_handling) == 0) {
```

Comment:
Are we sure these strings are always null-terminated at this point?

Reply:
Will change to ```strncmp``` just to be safe.